src/modules/shared/components/GenericTable/GenericTable.tsx

```diff
@@ -152,6 +152,10 @@ export const GenericTable = ({ classes, config }: GenericTableProps) => {
       }
     });
 
+    if (filteredArrFilters?.length) {
+      changePage(1);
```

Comment:
Essa condicional não iria ser triggada apenas quando o número de filtros fosse mudado? Se eu mudar o valor do filtro de nome=joao pra nome=roberto, o número de filtros não mudou, mas o número de resultados mudou, então deveria mudar a página também. Minha sugestão seria mudar para a página um sempre que esse useEffect for chamado (favor testar se isso não gera efeitos colaterais).